modules/p2/ding_entity_rating/ding_entity_rating.module

```diff
@@ -0,0 +1,473 @@
+    switch ($display['type']) {
+      case 'ding_entity_rating_action_default':
+        $element[$delta] = array(
+          '#markup' => theme('ding_entity_rating_display', array('entity' => $entity)),
+        );
+        break;
+    }
+  }
+
+  return $element;
+}
+
+/**
+ * Check if the current user has rated an entity.
+ *
+ * @param object $entity
+ *   The ting object entity to remove from the rating list.
+ *
+ * @return int
+ *   Rating if the user has rated the entity and FALSE if not.
+ */
+function ding_entity_rating_get_user_rating($entity) {
+  $list = ding_list_get_list(DING_ENTITY_RATING_LIST);
+  $result = FALSE;
+  $element = ding_list_has_ding_list_element($list, 'ting_object', $entity->ding_entity_id);
+
+  if ($element !== FALSE) {
+    $result = ding_base_get_value('ding_type', $element, 'field_weight');
+  }
+  
+  return $result;
+}
+
+/**
+ * Implements hook_theme().
+ */
+function ding_entity_rating_theme($existing, $type, $theme, $path) {
+  return array(
+    'ding_entity_rating_display' => array(
+      'variables' => array('entity' => NULL),
+    ),
+    'ding_entity_rating_average' => array(
+      'variables' => array('rating' => NULL),
+    ),
+    'ding_entity_rating_thank_you' => array(
+      'variables' => array(),
+    ),    
+    'ding_entity_rating_ding_user_complete_description' => array(
+      'variables' => array('history_list' => NULL, 'account' => NULL),
+    ),
+  );
+}
+
+/**
+ * Theme function for theme_ding_entity_rating_thank_you.
+ */
+function theme_ding_entity_rating_thank_you($variables) {
+  return '<div class="ding-entity-rating-respons">' . t('Thank you for your input') . '</div>';
+}
+
+/**
+ * Theme function for theme_ding_entity_rating_ding_user_complete_description.
+ */
+function theme_ding_entity_rating_ding_user_complete_description($variables) {
+  $uid = $variables['account']->uid;
+  $history_button = isset($variables['history_list']) ? "<li>" . l(t("Rate previous loans"), "list/" . $variables['history_list']->id) . "</li>" : "";
+  
+  return "<div class='description'>" .
+    t('You could rate materials to give us an idea of your preferences.') .
+    "</div><ul class='actions'><li>" . l(t("Rate your current loans"), "user/$uid/status/loans") . "</li>" .
+    $history_button .
+    "</ul>";
+}
+
+/**
+ * Preprocessor for ding_entity_rating_display.
+ */
+function template_preprocess_ding_entity_rating_display(&$variables) {
+  global $user;
+  $entity = $variables['entity'];
+  
+  // Add styling and javascript.
+  // Since we don't use render here, we attach them old fashioned way.
+  $module_path = drupal_get_path('module', 'ding_entity_rating');
+  drupal_add_css($module_path . '/css/ding_entity_rating.css');
+  drupal_add_js($module_path . '/js/ding_entity_rating.js');
+
+  $variables['rating'] = FALSE;
+  
+  // If the user hasn't rated and this isn't the rating list itself, enable
+  // rating.
+  $variables['classes_array'][] = 'ding-entity-rating';
```

Comment:
I do not understand the comment in relation to the code. The comment seems to infer some `if` statement but the code has none.